src/Chirp.Infrastructure/CheepRepository.cs

```diff
@@ -39,11 +36,31 @@ public IEnumerable<Cheep> QueryCheeps(int pageNumber, int amount, string? author
 
         if (!string.IsNullOrEmpty(author))
         {
-            queryResult = _db.Cheeps.Where(c => c.Author.Name == author);
+            _authorRepository.CreateAuthor(author, "example@mail.com");
+            var authorId = _authorRepository.FindAuthorsByName(author).First().AuthorId;
```

Comment:
This code could be optimised by just returning the newly created author from createAuthor.